### PR TITLE
Enable automation details for command base

### PR DESCRIPTION
**Pull request by eddynaka**
## Description

Imagine SARIF Pattern Matcher being run by someone, and we want to automatically ingest the output SARIF somehow.

With this, we could ask the user to run SARIF Pattern Matcher and pass `--automationId` and `--automationGuid`, that information would be stored in the SARIF and we would know that the SARIF was generated by a specific team.

## Follow-up change

Change the `commandBase` to accept a URI and this URI will be used to send the generated SARIF to that endpoint.

cc: @cfaucon

### CODE REVIEW

File: src/Sarif/Core/Run.cs [hunk 1]
```diff
@@ -220,6 +220,14 @@ public bool ShouldSerializeGraphs()
             return this.Graphs.HasAtLeastOneNonDefaultValue(Graph.ValueComparer);
         }
 
+        public bool ShouldSerializeAutomationDetails()
+        {
+            return this.AutomationDetails?.Description != null ||
```
Comment (michaelcfanning) [inline src/Sarif/Core/Run.cs:225]: >AutomationDetails?.Description [](http://example.com/codeflow?start=24&length=30)

Why should we serialize an empty description? suggest using is null or whitespace...

File: src/Sarif/Core/RunAutomationDetails.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// Information that describes a run's identity and role within an engineering system process.
+    /// </summary>
+    public partial class RunAutomationDetails
+    {
+        public bool ShouldSerializeDescription() => this.Description != null;
```
Comment (michaelcfanning) [inline src/Sarif/Core/RunAutomationDetails.cs:11]: >Description [](http://example.com/codeflow?start=57&length=11)

?? why persist an empty description? how is this useful?

File: src/Sarif.Driver/Sdk/AnalyzeOptionsBase.cs [hunk 3]
```diff
@@ -107,7 +107,17 @@ public abstract class AnalyzeOptionsBase : CommonOptionsBase
 
         [Option(
             "baseline",
-            HelpText = "A Sarif file to be used as baseline")]
+            HelpText = "A SARIF file to be used as baseline.")]
         public string BaselineSarifFile { get; set; }
+
+        [Option(
+            "automationId",
```
Comment (michaelcfanning) [inline src/Sarif.Driver/Sdk/AnalyzeOptionsBase.cs:114]: >automationId [](http://example.com/codeflow?start=13&length=12)

why aren't these in common options base? won't we need them for other commands? i guess we can move them later if so.

**Review by michaelcfanning — APPROVED** [review]
:shipit:

File: src/Sarif/Core/Run.cs (hunk 1 above)
Comment (eddynaka) [reply src/Sarif/Core/Run.cs:225]: The Description isn't a string, it is a Message type. It's why I didn't change to NullOrWhiteSpace check

File: src/Sarif.Driver/Sdk/AnalyzeOptionsBase.cs (hunk 3 above)
Comment (eddynaka) [reply src/Sarif.Driver/Sdk/AnalyzeOptionsBase.cs:114]: just moved! :)

File: src/Sarif/Core/RunAutomationDetails.cs (hunk 2 above)
Comment (eddynaka) [reply src/Sarif/Core/RunAutomationDetails.cs:11]: The Description isn't a string, it is a Message type. It's why I didn't change to NullOrWhiteSpace check

**Pull request merged**